daemon/oci_linux.go

```diff
@@ -314,9 +314,30 @@ func WithNamespaces(daemon *Daemon, c *container.Container) coci.SpecOpts {
 			setNamespace(s, ns)
 		}
 		// uts
-		if c.HostConfig.UTSMode.IsHost() {
+		if c.HostConfig.UTSMode.IsContainer() {
+			uc, err := daemon.getUtsContainer(c)
+			if err != nil {
+				return err
+			}
+			ns := specs.LinuxNamespace{
+				Type: "uts",
+				Path: fmt.Sprintf("/proc/%d/ns/uts", uc.State.GetPID()),
+			}
+			setNamespace(s, ns)
+			if userNS {
+				// to share a UTS namespace, they must also share a user namespace
+				nsUser := specs.LinuxNamespace{
+					Type: "user",
+					Path: fmt.Sprintf("/proc/%d/ns/user", uc.State.GetPID()),
+				}
+				setNamespace(s, nsUser)
+			}
```

Comment:
I see this is true for `pid` above -- is it definitely also true for `uts`?
(Sorry for not knowing -- my searching has failed me and I want to just make sure :sweat_smile:)

Reply:
It's not specifically true that any of the namespaces must share a user namespace, so I assumed this was an implementation opinion specific to docker, as all the other namespaces have the user namespace shared (ipc, net & pid).

I can remove it if you believe it wasn't intended to be an opinion on the implementation.